experimental/exporter/openmetrics/openmetrics.go

```diff
@@ -0,0 +1,181 @@
+package openmetrics
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/sdk/export"
+	"go.opentelemetry.io/sdk/metric/aggregator/counter"
+	"go.opentelemetry.io/sdk/metric/aggregator/gauge"
+	"go.opentelemetry.io/sdk/metric/aggregator/maxsumcount"
+)
+
+type (
+	Exporter struct {
+		dki dkiMap
+		agg aggMap
+		tmp [32]byte
+	}
+
+	aggEntry struct {
+		aggregator export.MetricAggregator
+		descriptor *export.Descriptor
+	}
+
+	dkiMap map[*export.Descriptor]map[core.Key]int
+	aggMap map[string]aggEntry
+)
+
+var _ export.MetricBatcher = &Exporter{}
+
+func NewExporter() *Exporter {
+	return &Exporter{
+		dki: dkiMap{},
+		agg: aggMap{},
+	}
+}
+
+func (e *Exporter) AggregatorFor(record export.MetricRecord) export.MetricAggregator {
+	switch record.Descriptor().MetricKind() {
+	case export.CounterMetricKind:
+		return counter.New()
+	case export.GaugeMetricKind:
+		return gauge.New()
+	case export.MeasureMetricKind:
+		return maxsumcount.New()
+	default:
+		panic("Unknown metric kind")
+	}
+}
+
+func (e *Exporter) Export(_ context.Context, record export.MetricRecord, agg export.MetricAggregator) {
+	desc := record.Descriptor()
+	keys := desc.Keys()
+
+	// Cache the mapping from Descriptor->Key->Index
+	ki, ok := e.dki[desc]
+	if !ok {
+		ki = map[core.Key]int{}
+		e.dki[desc] = ki
+
+		for i, k := range keys {
+			ki[k] = i
+		}
+	}
+
+	// TODO can add a cache of (LabelSet,Descriptor)->(Encoded).
+	// Can use reflection to build right-sized arrays, or impose a
+	// max-keys and use fixed-size arrays to speed this up.
+
+	// Compute the value list.  TODO: Unspecified values become
+	// empty strings, OK?
+	canon := make([]core.Value, len(keys))
+
+	for _, kv := range record.Labels() {
+		pos, ok := ki[kv.Key]
+		if !ok {
+			continue
+		}
+		canon[pos] = kv.Value
+	}
+
+	// Compute an encoded lookup key
+	var sb strings.Builder
+	for i := 0; i < len(keys); i++ {
+		sb.WriteString(string(keys[i]))
+		sb.WriteRune('=')
+
+		sb.WriteRune('"')
+		canon[i].Encode(&sb, e.tmp[:])
```

Comment:
We hopefully won't get a string value with a single unescaped double quotes.

Reply:
Just some initial work :)